### PR TITLE
Updating some verisons in order to be compatible with Symfony 4.2

### PR DESCRIPTION
**Pull request by victorlazov**

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Let's allow PHP 7.1 as well. This is what SF 4 requires.

**Review by ghost — CHANGES_REQUESTED** [review]
Check inspections. I found "use Symfony\Component\Validator\ValidatorInterface" - undefined class. Might be other problems as well.

**Comment by tomas7777** [comment]
Why is the support for "3.*" being dropped? "3.*" is being actively used in many projects.

**Comment by ghost** [comment]
> Why is the support for "3._" being dropped? "3._" is being actively used in many projects.

This library is rarely updated - has not been updated for about 7 months. If an update is needed for old versions of SF, maybe we can have a branch for them? Like for the old version 3.

**Comment by tomas7777** [comment]
It doesn't matter if it's rarely updated or not.
Making more branches doesn't make any sense in this case, because older symfony library support is backwards compatible (as it seems from your diff that no changes are needed), so there is no reason to drop support to them.

**Comment by victorlazov** [comment]
Dependency versions lowered.
Major version bump in order 2.x and 3.x support to be distinguished clearly.

File: composer.json [hunk 1]
```diff
@@ -7,11 +7,12 @@
...
     "require-dev": {
-        "phpunit/phpunit": "3.7.*"
+        "phpunit/phpunit": "^7.0"
     },
     "require": {
-        "symfony/property-access": "2.*|3.*",
-        "symfony/validator": "2.*|3.*",
+        "php": "^7.1",
+        "symfony/property-access": "^4.0",
+        "symfony/validator": "^4.0",
```
Comment (vbartusevicius) [inline composer.json]: please leave symfony 3.x support

File: composer.json [hunk 2]
```diff
@@ -7,11 +7,12 @@
         }
     },
     "require-dev": {
-        "phpunit/phpunit": "3.7.*"
+        "phpunit/phpunit": "^7.0"
     },
     "require": {
-        "symfony/property-access": "2.*|3.*",
-        "symfony/validator": "2.*|3.*",
+        "php": "^7.1",
```
Comment (vbartusevicius) [inline composer.json]: Please leave 7.0 support. I don't see where you use 7.1 features except `setUp` in tests. To have maximum backwards compatibility you should try to use lowest appropriate version.

**Review by vbartusevicius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (ghost) [reply composer.json]: Done

**Review by ghost — APPROVED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -7,9 +7,7 @@ env:
     - COMPOSER_NO_INTERACTION="1"
 
 php:
-  - '5.5'
-  - '5.6'
-  - '7.0'
```
Comment (vbartusevicius) [inline .travis.yml]: since your library is 7.0 compatible, please add 7.0 also

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,26 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 3.0.0
+### Changed
+```
+"php": "^7.1",
```
Comment (vbartusevicius) [inline CHANGELOG.md]: this block needs to be synced with composer.json

**Review by vbartusevicius — APPROVED** [review]

**Pull request merged**